nerfstudio/models/splatfactow.py

```diff
@@ -0,0 +1,1119 @@
+# ruff: noqa: E741
```

Comment:
it would be cool if this project used the nerfstudio-template ( https://github.com/nerfstudio-project/nerfstudio-method-template )

as does e.g. garfield ( https://github.com/chungmin99/garfield ) and the more recent dn_splatter ( https://github.com/maturk/dn-splatter )   

versus copy-pasta the entire splatfacto impl and hack stuff up just for one paper (as is done in this file).  it appears there are indeed a few parts of nerfstudio that need changing tho. highly recommend start as a template project and then maybe later port into main nerfstudio as newest splatfacto.  

while i do think down the line any gsplat user probably wants a good model that has a background model, should users really get pre-print material? or refereed, maybe even journal-quality, i.e. "de-facto".

Reply:
Yes I’m working on making this a template to make everything clearer. And probably need more tests on the viser side to make sure it won’t hurt other models.
Actually the first version of this works as a plugin of splatfacto but its implicit nature looks conflicted with the original gs. This makes everything messy.
As for the material wise for background model, I’m not very sure about if I should release the paper along with the code because it’s still unfinished and this is actually the first time I’ve tried to do a research project.
Thank you so much for your suggestion!